### ISSUE TITLE
[BUG] Plotting interactive confusion matrix doesn't work when using an eval_key in the evaluation 

### ISSUE THREAD

**Issue by leamasselles** [voxel51/fiftyone#3628]
### System information

-   **OS Platform and Distribution** (e.g., Linux Ubuntu 16.04): Ubuntu 22.04
-   **Python version** (`python --version`): Python 3.10.12
-   **FiftyOne version** (`fiftyone --version`): FiftyOne v0.21.6, Voxel51, Inc.
-   **FiftyOne installed from** (pip or source): pip

### Commands to reproduce

I am currently trying out Voxel51 with the tutorials on the website, especially the [interactive confusion matrix](https://docs.voxel51.com/user_guide/plots.html#confusion-matrices). I have been trying the different evaluation methods and then plotting them, it works fine until passing an `eval_key` parameter to `evaluate_detections`, `plot_confusion_matrix` then outputs `AttributeError: 'EvaluationPatchesView' object has no attribute '_DatasetView__name'`.

```python
import fiftyone as fo
import fiftyone.zoo as foz
from fiftyone import ViewField as F

dataset = foz.load_zoo_dataset("quickstart")

results = dataset.evaluate_detections(
    "predictions", gt_field="ground_truth", eval_key="eval"
)

# Get the 10 most common classes in the dataset
counts = dataset.count_values("ground_truth.detections.label")
classes = sorted(counts, key=counts.get, reverse=True)[:10]

session = fo.launch_app(dataset)

plot = results.plot_confusion_matrix(classes=classes)
plot.show(height=600)

session.plots.attach(plot, name="eval")
```

### Describe the problem

When executing this command in a Jupyter Notebook, the confusion matrix is plotted but is not interactive with the Voxel51 app. However, when writing:

```python
import fiftyone as fo
import fiftyone.zoo as foz
from fiftyone import ViewField as F

dataset = foz.load_zoo_dataset("quickstart")

# Evaluate detections in the `predictions` field
results = dataset.evaluate_detections("predictions", gt_field="ground_truth")

# The top-10 most common classes
counts = dataset.count_values("ground_truth.detections.label")
classes = sorted(counts, key=counts.get, reverse=True)[:10]

session = fo.launch_app(dataset)

# Plot confusion matrix
plot = results.plot_confusion_matrix(classes=classes)
plot.show(height=600)

session.plots.attach(plot)
```
The confusion matrix does get plotted and is interactive with the Voxel51 app, the difference being not having an `eval_key` parameter in `dataset.evaluate_detections()` and not giving a `name` value to `session.plots.attach()`

### Other info/logs

Here's the full stack trace, with my paths censored out:

```bash
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
voxel51/tiled_dataset_loader.ipynb Cell 20 line 6
      3 plot = results.plot_confusion_matrix(classes=classes)
      4 plot.show(height=600)
----> 6 session.plots.attach(plot, name="eval")

File voxel51/.env/lib/python3.10/site-packages/fiftyone/core/plots/manager.py:271, in PlotManager.attach(self, plot, name, connect, overwrite)
    269 else:
    270     self._connect_plot(name)
--> 271     self._update_plots([name])

File voxel51/.env/lib/python3.10/site-packages/fiftyone/core/plots/manager.py:666, in PlotManager._update_plots(self, names)
    663     self._update_view_plots(view_plot_names, view)
    665 for name in interactive_plot_names:
--> 666     self._update_interactive_plot(name, view)

File voxel51/.env/lib/python3.10/site-packages/fiftyone/core/plots/manager.py:714, in PlotManager._update_interactive_plot(self, name, view)
    712 elif plot.link_type == "labels":
    713     label_ids = self._get_current_label_ids_for_plot(plot)
--> 714     plot.select_ids(label_ids, view=view)
    715 else:
    716     raise ValueError(
    717         "InteractivePlot '%s' has unsupported link type '%s'"
    718         % (name, plot.link_type)
    719     )

File voxel51/.env/lib/python3.10/site-packages/fiftyone/core/plots/base.py:1107, in InteractivePlot.select_ids(self, ids, view)
   1104 if not self.is_connected:
   1105     return
-> 1107 self._select_ids(ids, view=view)

File voxel51/.env/lib/python3.10/site-packages/fiftyone/core/plots/plotly.py:2067, in InteractiveHeatmap._select_ids(self, ids, view)
   2065 def _select_ids(self, ids, view=None):
   2066     if ids is None:
-> 2067         self._deselect()
   2068         return
   2070     num_rows, num_cols = self.Z.shape

File voxel51/.env/lib/python3.10/site-packages/fiftyone/core/plots/plotly.py:2128, in InteractiveHeatmap._deselect(self)
   2125 self._curr_Z = None
   2126 self._curr_zlim = None
-> 2128 self._update_heatmap([], self.Z, self.Z, self.zlim)

File voxel51/.env/lib/python3.10/site-packages/fiftyone/core/plots/plotly.py:2165, in InteractiveHeatmap._update_heatmap(self, cells, Z_active, Z_bg, zlim)
   2162     self._bgw.zmax = zlim[1]
   2164 if self._selection_callback is not None:
-> 2165     self._selection_callback(self.selected_ids)

File voxel51/.env/lib/python3.10/site-packages/fiftyone/core/plots/manager.py:340, in PlotManager._connect_plot.<locals>._on_plot_selection(ids)
    339 def _on_plot_selection(ids):
--> 340     self._on_plot_selection(name, ids)

File voxel51/.env/lib/python3.10/site-packages/fiftyone/core/plots/manager.py:551, in PlotManager._on_plot_selection(self, name, ids)
    549 self._current_frame_ids = frame_ids
    550 self._current_labels = labels
--> 551 self._update_session(plot_view)
    552 self._update_plots_from_session(exclude=[name])

File voxel51/.env/lib/python3.10/site-packages/fiftyone/core/plots/manager.py:623, in PlotManager._update_session(self, view)
    621 self._last_session_view = view
    622 with self._session.no_show():
--> 623     self._session.view = view

File voxel51/.env/lib/python3.10/site-packages/fiftyone/core/session/session.py:256, in update_state.<locals>.decorator.<locals>.wrapper(session, *args, **kwargs)
    254 if auto_show and session.auto and focx.is_notebook_context():
    255     session.freeze()
--> 256 result = func(session, *args, **kwargs)
    257 session._client.send_event(StateUpdate(state=session._state))
    258 if auto_show and session.auto and focx.is_notebook_context():

File voxel51/.env/lib/python3.10/site-packages/fiftyone/core/session/session.py:693, in Session.view(self, view)
    687 if view is not None and not isinstance(view, fov.DatasetView):
    688     raise ValueError(
    689         "`Session.view` must be a %s or None; found %s"
    690         % (fov.DatasetView, type(view))
    691     )
--> 693 self._set_view(view)

File voxel51/.env/lib/python3.10/site-packages/fiftyone/core/session/session.py:711, in Session._set_view(self, view)
    708         self._set_dataset(view._root_dataset)
    710     self._state.view = view
--> 711     self._state.view_name = view.name
    713 self._state.selected = []
    714 self._state.selected_labels = []

File voxel51/.env/lib/python3.10/site-packages/fiftyone/core/view.py:291, in DatasetView.name(self)
    288 @property
    289 def name(self):
    290     """The name of the view if it is a saved view; otherwise None."""
--> 291     return self.__name

AttributeError: 'EvaluationPatchesView' object has no attribute '_DatasetView__name'
```

### What areas of FiftyOne does this bug affect?

-   [x] `App`: FiftyOne application issue
-   [x] `Core`: Core Python library issue
-   [ ] `Server`: FiftyOne server issue

### Willingness to contribute

The FiftyOne Community encourages bug fix contributions. Would you or another member of your organization be willing to contribute a fix for this bug to the FiftyOne codebase?

-   [ ] Yes. I can contribute a fix for this bug independently
-   [x] Yes. I would be willing to contribute a fix for this bug with guidance from the FiftyOne community
-   [ ] No. I cannot contribute a bug fix at this time

**Comment by brimoor** [comment]
Hi @leamasselles I believe this bug has been resolved in `fiftyone>=0.22`. Can you upgrade to the latest version and confirm that it works for you?

```
pip install --upgrade fiftyone
```

**Comment by leamasselles** [comment]
Hello, yes my bad, upgrading my Voxel51 and ipywidget versions fixed the issue, thanks for the help !
